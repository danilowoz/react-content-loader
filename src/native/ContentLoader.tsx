import * as React from 'react'
import { Circle, Rect } from 'react-native-svg'

import { Facebook, IContentLoaderProps } from '.'
import Svg from './Svg'

const ContentLoader: React.FC<IContentLoaderProps> = props =>
  props.children ? <Svg {...props} /> : <Facebook {...props} />

export { Circle, Rect }

export default ContentLoader
