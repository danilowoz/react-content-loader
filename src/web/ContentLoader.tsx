import * as React from 'react'

import { Facebook, IContentLoaderProps } from '.'
import Svg from './Svg'

const ContentLoader: React.FC<IContentLoaderProps> = props =>
  props.children ? <Svg {...props} /> : <Facebook {...props} />

export default ContentLoader
