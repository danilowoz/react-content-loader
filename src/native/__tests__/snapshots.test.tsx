import * as React from 'react'
import * as renderer from 'react-test-renderer'

import ContentLoader from '../ContentLoader'

describe('ContentLoader snapshots', () => {
  test('renders correctly the basic version', () => {
    const wrapper = renderer.create(
      <ContentLoader animate={false} uniqueKey="snapshots" />
    )
    const tree = wrapper.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders correctly with viewBox empty', () => {
    const wrapper = renderer.create(
      <ContentLoader animate={false} uniqueKey="snapshots" viewBox="" />
    )
    const tree = wrapper.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders correctly with viewBox defined', () => {
    const wrapper = renderer.create(
      <ContentLoader
        animate={false}
        uniqueKey="snapshots"
        viewBox="0 0 100 100"
      />
    )
    const tree = wrapper.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders correctly with viewBox defined and sizes defined too', () => {
    const wrapper = renderer.create(
      <ContentLoader
        animate={false}
        uniqueKey="snapshots"
        width={100}
        height={100}
        viewBox="0 0 100 100"
      />
    )
    const tree = wrapper.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
