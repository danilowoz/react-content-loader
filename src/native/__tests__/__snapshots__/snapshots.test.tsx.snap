// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentLoader snapshots renders correctly the basic version 1`] = `
<Svg
  height={124}
  style={Object {}}
  viewBox="0 0 476 124"
  width={476}
>
  <Rect
    clipPath="url(#snapshots-animated-diff)"
    fill="url(#snapshots-diff)"
    height="100%"
    width="100%"
    x="0"
    y="0"
  />
  <Defs>
    <ClipPath
      id="snapshots-animated-diff"
    >
      <Rect
        height="6"
        rx="3"
        width="88"
        x="48"
        y="8"
      />
      <Rect
        height="6"
        rx="3"
        width="52"
        x="48"
        y="26"
      />
      <Rect
        height="6"
        rx="3"
        width="410"
        x="0"
        y="56"
      />
      <Rect
        height="6"
        rx="3"
        width="380"
        x="0"
        y="72"
      />
      <Rect
        height="6"
        rx="3"
        width="178"
        x="0"
        y="88"
      />
      <Circle
        cx="20"
        cy="20"
        r="20"
      />
    </ClipPath>
    <LinearGradient
      id="snapshots-diff"
      x1="-100%"
      x2="100%"
      y1={0}
      y2={0}
    >
      <Stop
        offset={0}
        stopColor="#f5f6f7"
      />
      <Stop
        offset={0}
        stopColor="#eee"
      />
      <Stop
        offset={0}
        stopColor="#f5f6f7"
      />
    </LinearGradient>
  </Defs>
</Svg>
`;

exports[`ContentLoader snapshots renders correctly with viewBox defined 1`] = `
<Svg
  height={124}
  style={Object {}}
  viewBox="0 0 100 100"
  width={476}
>
  <Rect
    clipPath="url(#snapshots-animated-diff)"
    fill="url(#snapshots-diff)"
    height="100%"
    width="100%"
    x="0"
    y="0"
  />
  <Defs>
    <ClipPath
      id="snapshots-animated-diff"
    >
      <Rect
        height="6"
        rx="3"
        width="88"
        x="48"
        y="8"
      />
      <Rect
        height="6"
        rx="3"
        width="52"
        x="48"
        y="26"
      />
      <Rect
        height="6"
        rx="3"
        width="410"
        x="0"
        y="56"
      />
      <Rect
        height="6"
        rx="3"
        width="380"
        x="0"
        y="72"
      />
      <Rect
        height="6"
        rx="3"
        width="178"
        x="0"
        y="88"
      />
      <Circle
        cx="20"
        cy="20"
        r="20"
      />
    </ClipPath>
    <LinearGradient
      id="snapshots-diff"
      x1="-100%"
      x2="100%"
      y1={0}
      y2={0}
    >
      <Stop
        offset={0}
        stopColor="#f5f6f7"
      />
      <Stop
        offset={0}
        stopColor="#eee"
      />
      <Stop
        offset={0}
        stopColor="#f5f6f7"
      />
    </LinearGradient>
  </Defs>
</Svg>
`;

exports[`ContentLoader snapshots renders correctly with viewBox defined and sizes defined too 1`] = `
<Svg
  height={100}
  style={Object {}}
  viewBox="0 0 100 100"
  width={100}
>
  <Rect
    clipPath="url(#snapshots-animated-diff)"
    fill="url(#snapshots-diff)"
    height="100%"
    width="100%"
    x="0"
    y="0"
  />
  <Defs>
    <ClipPath
      id="snapshots-animated-diff"
    >
      <Rect
        height="6"
        rx="3"
        width="88"
        x="48"
        y="8"
      />
      <Rect
        height="6"
        rx="3"
        width="52"
        x="48"
        y="26"
      />
      <Rect
        height="6"
        rx="3"
        width="410"
        x="0"
        y="56"
      />
      <Rect
        height="6"
        rx="3"
        width="380"
        x="0"
        y="72"
      />
      <Rect
        height="6"
        rx="3"
        width="178"
        x="0"
        y="88"
      />
      <Circle
        cx="20"
        cy="20"
        r="20"
      />
    </ClipPath>
    <LinearGradient
      id="snapshots-diff"
      x1="-100%"
      x2="100%"
      y1={0}
      y2={0}
    >
      <Stop
        offset={0}
        stopColor="#f5f6f7"
      />
      <Stop
        offset={0}
        stopColor="#eee"
      />
      <Stop
        offset={0}
        stopColor="#f5f6f7"
      />
    </LinearGradient>
  </Defs>
</Svg>
`;

exports[`ContentLoader snapshots renders correctly with viewBox empty 1`] = `
<Svg
  height={124}
  style={Object {}}
  viewBox=""
  width={476}
>
  <Rect
    clipPath="url(#snapshots-animated-diff)"
    fill="url(#snapshots-diff)"
    height="100%"
    width="100%"
    x="0"
    y="0"
  />
  <Defs>
    <ClipPath
      id="snapshots-animated-diff"
    >
      <Rect
        height="6"
        rx="3"
        width="88"
        x="48"
        y="8"
      />
      <Rect
        height="6"
        rx="3"
        width="52"
        x="48"
        y="26"
      />
      <Rect
        height="6"
        rx="3"
        width="410"
        x="0"
        y="56"
      />
      <Rect
        height="6"
        rx="3"
        width="380"
        x="0"
        y="72"
      />
      <Rect
        height="6"
        rx="3"
        width="178"
        x="0"
        y="88"
      />
      <Circle
        cx="20"
        cy="20"
        r="20"
      />
    </ClipPath>
    <LinearGradient
      id="snapshots-diff"
      x1="-100%"
      x2="100%"
      y1={0}
      y2={0}
    >
      <Stop
        offset={0}
        stopColor="#f5f6f7"
      />
      <Stop
        offset={0}
        stopColor="#eee"
      />
      <Stop
        offset={0}
        stopColor="#f5f6f7"
      />
    </LinearGradient>
  </Defs>
</Svg>
`;
